examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+"""
+Title: Few-Shot learning with Reptile
+Author: [ADMoreau](https://github.com/ADMoreau)
+Date created: 2020/05/21
+Last modified: 2020/05/29
+Description: Reptile meta learning algorithm implemented in Keras
+
+"""
+
+"""
+The Reptile algorithm (https://arxiv.org/abs/1803.02999) was developed by OpenAI to
+perform model agnostic meta-learning. Specifically, this algorithm was designed to
+quickly learn to perform new tasks with minimal training, a task known as few-shot
+learning. The algorithm works by taking metagradients from the training of mini-batches
+representing new tasks, and using these gradients for a meta training step.
+"""
+
+import numpy as np
+import random
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow_datasets as tfds
+
+"""
+## Define the Parameters
+"""
+
+learning_rate = 0.003
+meta_step_size = 0.25
+
+inner_batch_size = 25
+eval_batch_size = 25
+
+meta_iters = 2000
+eval_iters = 5
+inner_iters = 4
+
+eval_interval = 1
+train_shots = 20
+shots = 5
+classes = 5
+
+"""
+## Dataset Functions
+"""
+
+
+class Dataset:
+    # This dataset function will fascilitate the creation of a few shot dataset
+    # from the Omniglot dataset that can be sampled from quickly while also
+    # allowing to create new labels at the same time.
+    def __init__(self, training):
+
+        # Download the tfrecord files containing the omniglot data and convert to a
+        # dataset.
+        if training:
```

Comment:
This class can generally be simplified quite a bit. One example would be these lines, where you only have to do:

```python
def __init__(self, split="train"):
   dataset = tfds.load("omniglot", split=split, as_supervised=True, shuffle_files=False)
```

Note that `batch(1)` is not useful since the dataset already yields samples one at a time.